rpc-server/src/utils.rs

```diff
@@ -194,6 +223,15 @@ pub enum ShadowDataConsistencyError {
     ResultsDontMatch(String),
 }
 
+/// Represents the rpc response status for data comparison.
+#[cfg(feature = "shadow_data_consistency")]
+#[derive(Debug, Clone)]
+pub enum ResponseState {
```

Comment:
Here, you mix the results of 2 responses in one object. The name does not explain this.
Later, you pass this state to `shadow_compare_results` while we haven't even seen the result of Archival RPC, we will make this request later inside this function. It's overcomplicated.

I suggest having only 2 options here - success and failure.
As a result, with these 2 options, not sure we need a separate structure here at all.